### ISSUE TITLE
Transient exception in getTransactionIsolation() permanently breaks pool

### ISSUE THREAD

**Issue by swankjesse** [brettwooldridge/HikariCP#887]
We had a temporary network hiccup while our service was starting up. The call to `connection.getTransactionIsolation()` failed with a SQLException and this log was printed:

```
2017-05-10 13:37:35,068 apa273.sjc2b.square franklin_us_east_ro connection adder franklin_us_east_ro - Default transaction isolation level detection failed (Communications link failure
```
We want to use the DB’s default isolation level so we don’t explicitly specify one in our configuration. But after this exception, the `transactionIsolation` is left at -1.

Later, when we actually attempt to exercise the connection, things fail because -1 isn’t a valid isolation level:

```
Caused by: java.sql.SQLException: Unsupported transaction isolation level '-1'
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:963)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:896)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:885)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:860)
	at com.mysql.jdbc.ConnectionImpl.setTransactionIsolation(ConnectionImpl.java:5122)
	at com.zaxxer.hikari.pool.PoolBase.setupConnection(PoolBase.java:406)
	... 10 more
```

I would prefer for the `PoolBase#checkDriverSupport()` call to not catch the SQLException. That way we don’t build a poisoned PoolBase that is never able to successfully connect to our DB.

**Comment by brettwooldridge** [comment]
Already fixed in HikariCP 2.6.1 (issue #821).

**Comment by swankjesse** [comment]
Thanks for the quick response. Sadly I’m already on 2.6.1 and still suffering this problem.
